client/ayon_core/host/interfaces/workfiles.py

```diff
@@ -0,0 +1,1214 @@
+
+        workdir_data = get_template_data(
+            project_entity,
+            folder_entity,
+            task_entity,
+            host_name=self.name,
+        )
+        workdir = get_workdir_with_workdir_data(
+            workdir_data,
+            project_name,
+            anatomy=anatomy,
+            template_key=template_key,
+            project_settings=project_settings,
+        )
+
+        if platform.system().lower() == "windows":
+            rootless_workdir = workdir.replace("\\", "/")
+        else:
+            rootless_workdir = workdir
+
+        used_roots = workdir.used_values.get("root")
+        if used_roots:
+            used_root_name = next(iter(used_roots))
+            root_value = used_roots[used_root_name]
+            workdir_end = rootless_workdir[len(root_value):].lstrip("/")
+            rootless_workdir = f"{{root[{used_root_name}]}}/{workdir_end}"
+
+        filenames = []
+        if os.path.exists(workdir):
+            filenames = list(os.listdir(workdir))
+
+        items = []
+        for filename in filenames:
+            filepath = os.path.join(workdir, filename)
+            # TODO add 'default' support for folders
+            ext = os.path.splitext(filepath)[1].lower()
+            if ext not in extensions:
+                continue
+
+            rootless_path = f"{rootless_workdir}/{filename}"
+            workfile_entity = workfile_entities_by_path.pop(
+                rootless_path, None
+            )
+            items.append(WorkfileInfo.new(
+                filepath,
+                rootless_path,
+                available=True,
+                workfile_entity=workfile_entity,
+            ))
+
+        for workfile_entity in workfile_entities_by_path.values():
+            # Workfile entity is not in the filesystem
+            #   but it is in the database
+            rootless_path = workfile_entity["path"]
+            ext = os.path.splitext(rootless_path)[1].lower()
+            if ext not in extensions:
+                continue
+            filepath = anatomy.fill_root(rootless_path)
+            items.append(WorkfileInfo.new(
+                filepath,
+                rootless_path,
+                available=False,
+                workfile_entity=workfile_entity,
+            ))
+
+        return items
+
+    def list_published_workfiles(
+        self,
+        project_name: str,
+        folder_id: str,
+        *,
+        anatomy: Optional["Anatomy"] = None,
+        version_entities: Optional[list[dict[str, Any]]] = None,
+        repre_entities: Optional[list[dict[str, Any]]] = None,
+    ) -> list[PublishedWorkfileInfo]:
+        """List published workfiles for the given folder.
+
+        The default implementation looks for products with the 'workfile'
+            product type.
+
+        Pre-fetched entities have mandatory fields to be fetched.
+         -  Version: 'id', 'author', 'taskId'
+         -  Representation: 'id', 'versionId', 'files'
```

Comment:
Just so the docstring generator will handle this.

```suggestion
        Pre-fetched entities have mandatory fields to be fetched:
            - Version: 'id', 'author', 'taskId'
            - Representation: 'id', 'versionId', 'files'
```